### PR TITLE
Add `binary-bitwise-expression` spec conformance tests

### PR DESCRIPTION
**Pull request by rdulmina**
## Purpose
$title

Related lang issue [#32465](https://github.com/ballerina-platform/ballerina-lang/issues/32465)

### CODE REVIEW

File: conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt [hunk 1]
```diff
@@ -0,0 +1,923 @@
...
+
+    io:println(g & g); // @output 54522
+    io:println(g & h); // @output 250
+
+    io:println(h & h); // @output 255
+}
+
+Test-Case: output
+Description: Test bitwise and expression evaluates to nil when either operand is nil.
+Label: binary-bitwise-expr, byte, int:Signed8, int:Signed16, int:Signed32, int:Unsigned8, int:Unsigned16,
```
Comment (lasinicl) [inline conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt:346]: missing `nil-literal` label?

File: conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt [hunk 2]
```diff
@@ -0,0 +1,923 @@
...
+
+public function main() {
+    MyType a = 5;
+    MyType b = 433;
+    int c = 765;
+
+    printResult(a & b); // @output 1
+    printResult(a & 12345); // @output 1
+    printResult(b & c); // @output 177
+    printResult(a & c); // @output 5
```
Comment (lasinicl) [inline conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt:436]: Shall we also add some tests with user-defined types such as `type MyType int:Signed16|int:Unsigned32` and unions like `int:Signed16|int:Unsigned32`? Check with nillable types and other `binary-bitwise-expr` as well.

File: conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_or_expr.balt [hunk 3]
```diff
@@ -0,0 +1,966 @@
...
+
+    int:Unsigned16 n = 12;
+    int:Unsigned16 o = 4434;
+    int:Signed16 _ = n | o; // @error the smallest suck K is 16 hence, the static type of the result is Unsigned16
+
+    int:Unsigned32 q = 12;
+    int:Unsigned16 r = 4434;
+    int:Unsigned8 _ = q | r; // @error the smallest suck K is 32 hence, the static type of the result is Unsigned32
+}
+
```
Comment (lasinicl) [inline conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_or_expr.balt:651]: Shall we add some using `binary-bitwise-expr` at module level?

Comment (rdulmina) [reply conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_or_expr.balt:651]: will add

File: conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt (hunk 1 above)
Comment (rdulmina) [reply conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt]: thanx

File: conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt (hunk 2 above)
Comment (rdulmina) [reply conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt]: Added

File: conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt [hunk 4]
```diff
@@ -0,0 +1,1314 @@
+Test-Case: output
+Description: Test basic bitwise and expression.
+Label: binary-bitwise-expr
```
Comment (MaryamZi) [inline conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt:3]:
```suggestion
Labels: binary-bitwise-expr
```

Let's update everywhere.

Comment (MaryamZi) [inline conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt:3]: Need to include the int label. Please update wherever applicable.

**Review by MaryamZi — COMMENTED** [review]
There seem to be multiple test failures.

File: conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_or_expr.balt [hunk 5]
```diff
@@ -0,0 +1,1456 @@
...
+    int:Signed32 c = 5454545;
+    int:Signed32 d = -341345678;
+
+    io:println(a | b); // @output 20
+    io:println(b | a); // @output 20
+    io:println(b | c); // @output 16
+    io:println(c | b); // @output 16
+    io:println(c | c); // @output 5454545
+    io:println(d | d); // @output -341345678
+    io:println(c | d); // @output 211536
```
Comment (lochana-chathura) [inline conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_or_expr.balt:133]: outputs seem to be wrong except for L131 and L132.

File: conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt [hunk 6]
```diff
@@ -0,0 +1,1317 @@
...
+
+function init() {
+    printResult((54 << 1) & (534 << 2)); // @output 72
+    printResult(54 << 1 & 534 << 2); // @output 72
+
+    printResult((54 >> 1) & (534 >> 1)); // @output 11
+    printResult(54 >> 1 & 534 >> 1); // @output 11
+
+    printResult((-54 >>> 1) & (-534 >>> 1)); // @output 2147483365
+    printResult(-54 >>> 1 & -534 >>> 1); // @output 2147483365
```
Comment (lochana-chathura) [inline conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt:1243]: Outputs seem to be wrong

File: conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_xor_expr.balt [hunk 7]
```diff
@@ -0,0 +1,1462 @@
...
+}
+
+Test-Case: output
+Description: Test basic bitwise xor expression with negative integer values.
+Labels: binary-bitwise-expr, DecimalNumber, unary-minus
+
+function init() {
+    io:println(-89 ^ 6); // @output -95
+    io:println(6 ^ -9); // @output -15
+    io:println(-43554 ^ -6); // @output -43556
```
Comment (lochana-chathura) [inline conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_xor_expr.balt:34]:
```suggestion
    io:println(-43554 ^ -6); // @output 43556
```

File: conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_xor_expr.balt [hunk 8]
```diff
@@ -0,0 +1,1462 @@
...
+    int:Signed16 b = 987;
+    int:Signed16 c = 7764;
+    int:Signed16 d = -9876;
+
+    io:println(a ^ b); // @output 387953143
+    io:println(b ^ a); // @output 387953143
+    io:println(b ^ c); // @output 7567
+    io:println(c ^ b); // @output 7567
+    io:println(c ^ c); // @output 0
+    io:println(d ^ d); // @output -0
```
Comment (lochana-chathura) [inline conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_xor_expr.balt:154]:
```suggestion
    io:println(d ^ d); // @output 0
```

File: conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_xor_expr.balt [hunk 9]
```diff
@@ -0,0 +1,1462 @@
...
+    int:Signed8 b = 97;
+    int:Signed8 c = 64;
+    int:Signed8 d = -23;
+
+    io:println(a ^ b); // @output 387953229
+    io:println(b ^ a); // @output 387953229
+    io:println(b ^ c); // @output 33
+    io:println(c ^ b); // @output 33
+    io:println(c ^ c); // @output 0
+    io:println(d ^ d); // @output -0
```
Comment (lochana-chathura) [inline conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_xor_expr.balt:173]:
```suggestion
    io:println(d ^ d); // @output 0
```

File: conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_xor_expr.balt [hunk 10]
```diff
@@ -0,0 +1,1462 @@
...
+const VALUE1 = 5;
+const VALUE2 = 43433;
+
+const int VALUE4 = 654454;
+const byte VALUE5 = 65;
+
+function init() {
+    int value3 = -9876533;
+
+    printResult(VALUE1 ^ VALUE2); // @output 43437
```
Comment (lochana-chathura) [inline conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_xor_expr.balt:1189]:
```suggestion
    printResult(VALUE1 ^ VALUE2); // @output 43436
```

File: conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt [hunk 11]
```diff
@@ -0,0 +1,1317 @@
...
+    int:Unsigned16 d = 4344;
+    int:Unsigned16 e = b & d;
+    io:println(e); // @output 4168
+
+    int:Unsigned8 f = 44;
+    int:Unsigned8 g = d & f;
+    io:println(g); // @output 40
+
+    int:Unsigned8 h = f & g;
+    io:println(h); // @output 40
```
Comment (lochana-chathura) [inline conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt:422]: both operands being int:Unsigned8 is covered in the test above. Shall we change this to cover Unsigned32 & Unsigned8 combination?

File: conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt [hunk 12]
```diff
@@ -0,0 +1,1317 @@
...
+
+    byte i = f & g;
+    io:println(i); // @output 40
+}
+
+Test-Case: output
+Description: Test if the type of either operand is a subtype of int:UnsignedK when K is 8, 16 or 32, then the static
+             type of the result is int:UnsignedN where N is the smallest such K via inference.
+Labels: binary-bitwise-expr, DecimalNumber, int, int:Unsigned8, int:Unsigned16, int:Unsigned32, var
```
Comment (lochana-chathura) [inline conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt:431]: label for byte

File: conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt [hunk 13]
```diff
@@ -0,0 +1,1317 @@
...
+
+    var i = f & g;
+    byte result5 = h;
+    io:println(result5); // @output 40
+}
+
+Test-Case: error
+Description: Test if the type of either operand is a subtype of int:UnsignedK when K is 8, 16 or 32, then the static
+             type of the result is int:UnsignedN where N is the smallest such K using invalid assignment.
+Labels: binary-bitwise-expr, DecimalNumber, int, int:Unsigned8, int:Unsigned16, int:Unsigned32, int:Signed8, int:Signed32
```
Comment (lochana-chathura) [inline conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt:462]: label for byte

File: conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt [hunk 14]
```diff
@@ -0,0 +1,1317 @@
...
+    int:Unsigned32 q = 12;
+    int:Unsigned16 r = 4434;
+    int:Unsigned8 _ = q & r; // @error the static type of the result is Unsigned16
+}
+
+Test-Case: error
+Description: Test if the type of either operand is a subtype of int:UnsignedK when K is 8, 16 or 32, then the static
+             type of the result is int:UnsignedN where N is the smallest such K via invalid assignment with inference.
+Labels: binary-bitwise-expr,  DecimalNumber, int, int:Unsigned8, int:Unsigned16, int:Unsigned32, int:Signed8, 
+        int:Signed32, var
```
Comment (lochana-chathura) [inline conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt:496]: label for byte, `int:Signed16`. 
seems `int:Signed32` is redundant.

File: conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt [hunk 15]
```diff
@@ -0,0 +1,1317 @@
...
+    printResult(h & h); // @output 255
+}
+
+function printResult(int? result) {
+    io:println(result);
+}
+
+Test-Case: output
+Description: Test bitwise and expression evaluates to nil when either operand is nil.
+Labels: binary-bitwise-expr, byte,  DecimalNumber, int, int:Signed8, int:Signed16, int:Signed32, int:Unsigned8,
```
Comment (lochana-chathura) [inline conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt:590]:
```suggestion
Labels: binary-bitwise-expr, byte, DecimalNumber, int, int:Signed8, int:Signed16, int:Signed32, int:Unsigned8, 
```

File: conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt [hunk 16]
```diff
@@ -0,0 +1,1317 @@
...
+    int:Signed8? e = 100;
+
+    int:Unsigned32? f = 21474836;
+    int:Unsigned16? g = 54522;
+    int:Unsigned8? h = 255;
+
+    int? i = ();
+    byte? j = ();
+    int:Signed32? k = ();
+    int:Unsigned16? l = ();
```
Comment (lochana-chathura) [inline conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt:609]: usually we don't use letter like `l` for variable names as it is confusing with 1. 
```suggestion
    int:Unsigned16? m = ();
```

File: conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt [hunk 17]
```diff
@@ -0,0 +1,1317 @@
...
+
+    printResult((a & i)); // @output ()
```
Comment (lochana-chathura) [inline conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt:611]: do we need these extra parenthesis?
```suggestion
    printResult(a & i); // @output ()
```

File: conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt [hunk 18]
```diff
@@ -0,0 +1,1317 @@
...
+    io:println(result4); // @output 40
+
+    int i = 45343;
+    int? j = 5454;
+    int? result5 = i & j;
+    io:println(result5); // @output 4366
+
+    int l = 5454545343;
+    byte? m = 43;
+    byte? result6 = l & m;
```
Comment (lochana-chathura) [inline conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt:703]: Here also. Let's check all places where we have used `l`.

File: conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt [hunk 19]
```diff
@@ -0,0 +1,1317 @@
...
+    printResult((l & l)); // @output ()
+}
+
+function printResult(int? result) {
+    io:println(result.toBalString());
+}
+
+Test-Case: output
+Description: Test the static type of bitwise and expression will be a nillable type when at least one operand is nillable.
+Labels: binary-bitwise-expr, byte,  DecimalNumber, int, int:Signed8, int:Signed16, int:Signed32, int:Unsigned8,
```
Comment (lochana-chathura) [inline conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt:672]: No usage of int:Signed8, int:Signed16, int:Signed32 within the test

File: conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt [hunk 20]
```diff
@@ -0,0 +1,1317 @@
...
+    int o = 5454545343;
+    byte? p = ();
+    int:Unsigned8? result7 = o & p;
+    io:println(result7.toBalString()); // @output ()
+}
+
+Test-Case: error
+Description: Test the static type of bitwise and expression will be a nillable type when at least one operand is nillable
+             via inference and invalid assignments.
+Labels: binary-bitwise-expr, byte,  DecimalNumber, int, int:Signed8, int:Signed16, int:Signed32, int:Unsigned8,
```
Comment (lochana-chathura) [inline conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt:715]: No usage of int:Signed8, int:Signed16, int:Signed32

File: conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt [hunk 21]
```diff
@@ -0,0 +1,1317 @@
...
+    byte _ = l & m; // @error the static type of the result is byte?
+
+    int o = 5454545343;
+    byte? p = ();
+    int:Unsigned8 _ = o & p; // @error the static type of the result is byte?
+}
+
+Test-Case: output
+Description: Test bitwise and expression at module level.
+Labels: binary-bitwise-expr, byte, DecimalNumber, int, int:Signed8, int:Signed16, int:Signed32, int:Unsigned8,
```
Comment (lochana-chathura) [inline conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt:750]: Seems we need a label for `module-init-var-decl`

File: conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt [hunk 22]
```diff
@@ -0,0 +1,1317 @@
...
+    printResult(result35); // @output 255
+}
+
+function printResult(int? result) {
+    io:println(result);
+}
+
+Test-Case: output
+Description: Test bitwise and expression where the operands are user defined type T where the T is a subtype of int.
+Labels: binary-bitwise-expr, DecimalNumber, int, module-type-defn, int:Signed8, int:Signed32, int:Unsigned16,
```
Comment (lochana-chathura) [inline conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt:968]: label for union-type

File: conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt [hunk 23]
```diff
@@ -0,0 +1,1317 @@
...
+    printResult(VALUE4 & VALUE4); // @output 654454
+}
+
+function printResult(int result) {
+    io:println(result);
+}
+
+Test-Case: output
+Description: Test float literal as operands with type cast.
+Labels: binary-bitwise-expr, DecimalNumber, DecimalFloatingPointNumber, int, float, type-cast-expr
```
Comment (lochana-chathura) [inline conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt:1063]: label for any

File: conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt [hunk 24]
```diff
@@ -0,0 +1,1317 @@
...
+    printResult(<int> f & c); // @output 1042
+}
+
+function printResult(int result) {
+    io:println(result);
+}
+
+Test-Case: output
+Description: Test field access and optional field access expression as operands.
+Labels: binary-bitwise-expr, DecimalNumber, int, module-type-defn, field-access-expr, numeric-nil-lifting, optional-field-access-expr,
```
Comment (lochana-chathura) [inline conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt:1088]: label for mapping-constructor

File: conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt [hunk 25]
```diff
@@ -0,0 +1,1317 @@
...
+    printResult(p1.id & p2?.age); // @output 1
+}
+
+function printResult(int? result) {
+    io:println(result);
+}
+
+Test-Case: output
+Description: Test member access expression as operands.
+Labels: binary-bitwise-expr, DecimalNumber, int, member-access-expr
```
Comment (lochana-chathura) [inline conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt:1113]: label for array-type

File: conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt [hunk 26]
```diff
@@ -0,0 +1,1317 @@
...
+        return error("No data");
+    }
+}
+
+function printResult(int result) {
+    io:println(result);
+}
+
+Test-Case: parser-error
+Description: Test syntax errors in bitwise and expression
```
Comment (lochana-chathura) [inline conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt:1311]:
```suggestion
Description: Test syntax errors in bitwise and expression.
```

**Comment by hasithaa** [comment]
The Team is busy with finalizing the SwanLake GA release. Hence closing this PR temporarily.

File: conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt (hunk 17 above)
Comment (rdulmina) [reply conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt:611]: No :-D

File: conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt [hunk 27]
```diff
@@ -0,0 +1,1330 @@
...
+
+    int d = 9876;
+    int:Unsigned32 e = 5454;
+    var result2 = d & e;
+    int:Unsigned16 _ = result2; // @error smallest K of the operands is 32
+
+    int g = 9876;
+    int:Unsigned16 h = 5454;
+    var result3 = g & h;
+    int:Unsigned8 _ = result3; // @error smallest K of the operands is 16
```
Comment (MaryamZi) [inline conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt:513]: The idea for error tests is to be able to compare this text and the actual error message and validate that it is indeed the correct error message. Shall we rethink these errors at least similar to L503?

File: conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt [hunk 28]
```diff
@@ -0,0 +1,1330 @@
...
+    printResult(g & h); // @output 250
+
+    printResult(h & h); // @output 255
+}
+
+function printResult(int? result) {
+    io:println(result);
+}
+
+Test-Case: output
```
Comment (MaryamZi) [inline conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt:589]: Don't we have error test for the static type with numeric nil lifting?

Comment (rdulmina) [reply conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt:590]: Yes, we have. https://github.com/ballerina-platform/ballerina-spec/pull/941/files#diff-b61d4efe62f7b5e3b7b97e8277d5f19fa183e9101e70b39fb6e13b88c68c1e5dR716

Comment (rdulmina) [reply conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt:590]: L#716

File: conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_or_expr.balt [hunk 29]
```diff
@@ -0,0 +1,1365 @@
...
+    printResult(g | h); // @output 75
+}
+
+function printResult(int result) {
+    io:println(result);
+}
+
+Test-Case: output
+Description: Test bitwise or expression where the operands are user-defined type T where the T is a subtype of int.
+Fail-Issue: ballerina-platform/ballerina-lang#32888
```
Comment (lochana-chathura) [inline conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_or_expr.balt:1040]:
```suggestion
Fail-Issue: ballerina-platform/ballerina-lang#35247
```

File: conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_xor_expr.balt [hunk 30]
```diff
@@ -0,0 +1,1342 @@
...
+    printResult(((4341 & 1) ^ 234) | 1); // @output 235
+}
+
+function printResult(int result) {
+    io:println(result);
+}
+
+Test-Case: output
+Description: Test check expression as operands.
+Labels: binary-bitwise-expr, check, checkpanic, DecimalNumber, error, int, optional-type
```
Comment (lochana-chathura) [inline conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_xor_expr.balt:1320]:
```suggestion
Labels: binary-bitwise-expr, check, checkpanic, DecimalNumber, error-type, int, optional-type, if-else-stmt
```

File: conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_or_expr.balt [hunk 31]
```diff
@@ -0,0 +1,1339 @@
...
+    return 99.4;
+}
+
+function getFloat2() returns float {
+    return 43.8;
+}
+
+Test-Case: error
+Description: Test the static type of both operands must be a subtype of int via invalid other operand types.
+Labels: binary-bitwise-expr, list-constructor-expr, mapping-constructor-expr, map-type, string
```
Comment (lochana-chathura) [inline conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_or_expr.balt:321]:
```suggestion
Labels: binary-bitwise-expr, list-constructor-expr, mapping-constructor-expr, map-type, string, nil-literal
```

File: conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_or_expr.balt [hunk 32]
```diff
@@ -0,0 +1,1339 @@
...
+    printResult((4341 & 1) | (234 ^ 1)); // @output 235
+}
+
+function printResult(int result) {
+    io:println(result);
+}
+
+Test-Case: output
+Description: Test check expression as operands.
+Labels: binary-bitwise-expr, check, checkpanic, DecimalNumber, error, int, optional-type
```
Comment (lochana-chathura) [inline conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_or_expr.balt:1317]:
```suggestion
Labels: binary-bitwise-expr, check, checkpanic, DecimalNumber, error-type, int, optional-type, if-else-stmt
```

File: conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt [hunk 33]
```diff
@@ -0,0 +1,1339 @@
...
+    printResult(4341 | ((1 & 234) ^ 1)); // @output 4341
+}
+
+function printResult(int result) {
+    io:println(result);
+}
+
+Test-Case: output
+Description: Test check expression as operands.
+Labels: binary-bitwise-expr, check, checkpanic, DecimalNumber, error, int, optional-type
```
Comment (lochana-chathura) [inline conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt:1317]:
```suggestion
Labels: binary-bitwise-expr, check, checkpanic, DecimalNumber, error-type, int, optional-type, if-else-stmt
```

**Review by lochana-chathura — APPROVED** [review]
"Test bitwise or expression where the operands are user-defined type T where the T is a subtype of int." failing in bitwise_or as the spec branch is not synced with latest 2201.0.x. Syncing locally passed the tests. Hence approving.

File: conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt [hunk 34]
```diff
@@ -0,0 +1,1338 @@
...
+}
+
+function printResult(int? result) {
+    io:println(result);
+}
+
+Test-Case: output
+Description: Test bitwise and expression where the operands are user-defined type T where the T is a subtype of int.
+Labels: binary-bitwise-expr, byte, int, module-type-defn, int:Signed8, int:Signed32, int:Unsigned16, 
+        int:Unsigned32, singleton-type, unary-minus, union-type
```
Comment (MaryamZi) [inline conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt:988]: Re: singletons and user-defined types, shouldn't the following work?

```ballerina
type X 1|2;
type Y 256|3000;

public function main() {
    X a = 1;
    Y b = 256;

    int:Unsigned8 _ = a & b;
}
```

Doesn't work atm. We don't have such tests, do we?

Comment (rdulmina) [reply conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt:988]: Thanx for mentioning it. We don't have such tests. will create an issue and add tests

Comment (rdulmina) [reply conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt:1016]: Tests were added and created issue #35458

**Review by gimantha — APPROVED** [review]

**Pull request merged**